apps/opencs/view/render/pathgrid.cpp

```diff
@@ -0,0 +1,547 @@
+#include "pathgrid.hpp"
+
+#include <algorithm>
+
+#include <osg/Geode>
+#include <osg/Geometry>
+#include <osg/Group>
+#include <osg/PositionAttitudeTransform>
+#include <osg/Vec3>
+
+#include <components/sceneutil/pathgridutil.hpp>
+
+#include "../../model/world/cell.hpp"
+#include "../../model/world/commands.hpp"
+#include "../../model/world/commandmacro.hpp"
+#include "../../model/world/data.hpp"
+#include "../../model/world/idtree.hpp"
+
+namespace CSVRender
+{
+    class PathgridNodeCallback : public osg::NodeCallback
+    {
+        public:
+
+            virtual void operator()(osg::Node* node, osg::NodeVisitor* nv)
+            {
+                PathgridTag* tag = dynamic_cast<PathgridTag*>(node->getUserData());
+                tag->getPathgrid()->update();
+            }
+    };
+
+    PathgridTag::PathgridTag(Pathgrid* pathgrid)
+        : TagBase(Mask_Pathgrid), mPathgrid(pathgrid)
+    {
+    }
+
+    Pathgrid* PathgridTag::getPathgrid() const
+    {
+        return mPathgrid;
+    }
+
+    QString PathgridTag::getToolTip(bool hideBasics) const
+    {
+        QString text("Pathgrid: ");
+        text += mPathgrid->getId().c_str();
+
+        if (!hideBasics)
```

Comment:
Better remove this, I think that empty if statements might trigger warnings on some static analysis tools.